sdk/resources/Microsoft.Azure.Management.Resource/src/Generated/Models/GenericResource.cs

```diff
@@ -44,8 +45,8 @@ public GenericResource()
         /// resource.</param>
         /// <param name="sku">The SKU of the resource.</param>
         /// <param name="identity">The identity of the resource.</param>
-        public GenericResource(string id = default(string), string name = default(string), string type = default(string), string location = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), Plan plan = default(Plan), object properties = default(object), string kind = default(string), string managedBy = default(string), Sku sku = default(Sku), Identity identity = default(Identity))
-            : base(id, name, type, location, tags)
+        public GenericResource(string id = default(string), string name = default(string), string type = default(string), string location = default(string), ExtendedLocation extendedLocation = default(ExtendedLocation), IDictionary<string, string> tags = default(IDictionary<string, string>), Plan plan = default(Plan), object properties = default(object), string kind = default(string), string managedBy = default(string), Sku sku = default(Sku), Identity identity = default(Identity))
+            : base(id, name, type, location, extendedLocation, tags)
```

Comment:
Breaking change.  Please add a customization to mitigate

Reply:
Maybe I misunderstood something, but since 2021-01-01 is a new API version and this is the first time we generate SDK for it, is adding a new property still considered a breaking change? If so, could you please give me some pointers on how to add a customization?

Reply:
Yes, a breaking change is a breaking change.  You are not allowed to have breaking changes from the previous stable release without breaking change approval.  Even with breaking change approval, it is best for customers to avoid them, and this is a breaking change that is very minimal effort to avoid.  In this case, since the models are all partial classes, you can just add a customization in a partial class in the customizations folder that implements the old constructor signature:

``` c#
public GenericResource(string id, string name, string type, string location, IDictionary<string, string> tags, Plan plan = default(Plan), object properties = default(object), string kind = default(string), string managedBy = default(string), Sku sku = default(Sku), Identity identity = default(Identity))
```

This will ensure that any existing usage of the parameters will still be supported.

Reply:
I see. Thanks for the explanation! I'll add customizations.

Reply:
You can (should) even call the new constructor in the implementation, this should just be a few lines of code.

Reply:
Yeah good idea. Added a customization that implements the old constructor signature by calling the new constructor.